hazelcast/src/main/java/com/hazelcast/datalink/HazelcastDataLink.java

```diff
@@ -77,25 +86,16 @@ public void shutdown() {
     }
 
     private ClientConfig buildClientConfig() {
-        ClientConfig clientConfig = null;
-        String clientXml = getConfig().getProperty(CLIENT_XML);
-        if (!StringUtil.isNullOrEmpty(clientXml)) {
-            // Read ClientConfig from XML
-            clientConfig = asClientConfig(clientXml);
-        }
+        DataLinkConfig dataLinkConfig = getConfig();
 
-        String clientYaml = getConfig().getProperty(CLIENT_YML);
-        if (!StringUtil.isNullOrEmpty(clientYaml)) {
-            // Read ClientConfig from Yaml
-            clientConfig = asClientConfigFromYaml(clientYaml);
-        }
+        HazelcastDataLinkConfigValidator validator = new HazelcastDataLinkConfigValidator();
+        validator.validate(dataLinkConfig);
 
-        if (clientConfig == null) {
-            throw new HazelcastException("HazelcastDataLink with name '" + getConfig().getName()
-                    + "' could not be created, provide either client_xml or client_yml property "
-                    + "with the client configuration.");
-        }
-        return clientConfig;
+        HazelcastDataLinkConfigLoader configLoader = new HazelcastDataLinkConfigLoader();
+        configLoader.loadConfigFromFile(dataLinkConfig);
```

Comment:
It's best we treat the config as immutable. Could you restructure the code so it doesn't set the property.

Reply:
rev : 0388e0c95f24c270782fe98f46384f94a6b3b03d